src/Components/Navbar/Navbar.jsx

```diff
@@ -6,18 +6,88 @@ import { useState } from "react";
 import { useProductList } from "../../context/ProductListContext";
 import { get_searched_result } from "../../utils";
 import { useNavigate } from "react-router-dom";
+import { useAuthContext } from "../../context/AuthContext";
+import { useToast } from "../../context/ToastContext";
 
 export const Navbar = () => {
   useGetLocalData();
-  const { user_data } = useUserData();
+  const { handleaddtoast } = useToast();
+  const { user_data, setUser_Data } = useUserData();
   const [isInputActive, setIsInputActive] = useState(false);
   const [searchValue, setSearchValue] = useState("");
   const { product_list } = useProductList();
   const searched_result = get_searched_result(searchValue, product_list);
+  const [sideBarActive, setSideBarActive] = useState(false);
   let navigate = useNavigate();
+  const { auth_state, setAuthState } = useAuthContext();
+  const { token, user } = auth_state;
+
   return (
     <>
       <div className="navigation-bar z-ind-2">
+        <div
+          onClick={() => {
+            setSideBarActive((prevState) => !prevState);
+          }}
+          className="hemburgur"
+          href="#"
+        >
+          <i className="fas fa-bars"></i>
+        </div>
+        {sideBarActive && (
+          <div className="side-hidden-bar  z-ind-2">
+            <div className="user-info pad-2 flex-align jst-sp-sb">
+              <img
+                className="avatar avatar-md"
+                src="https://cdn.pixabay.com/photo/2016/08/08/09/17/avatar-1577909_1280.png"
+                alt="avatar-image"
+              />
+              <div
+                onClick={() => {
+                  setSideBarActive((prevState) => !prevState);
+                }}
+                className="cross cursor-pointer"
+              >
+                <i className="fas fa-times"></i>
+              </div>
+            </div>
+            {token ? (
+              <div className="side-hidder-info pad-t-1 pad-l-1">
+                <div
+                  onClick={() => {
+                    handleaddtoast({
+                      message: `Bye Bye ${user.firstName}`,
+                      type: "alert-success",
+                    });
+                    localStorage.removeItem("token");
+                    localStorage.removeItem("my_wishlist");
+                    localStorage.removeItem("my_cart");
+                    setUser_Data({ type: "RESET" });
+
+                    setAuthState({ type: "TOKEN", payload: null });
+                    setSideBarActive((pre_state) => !pre_state);
+                  }}
+                  className="btn btn-primary cursor-pointer"
+                >
+                  LogOut
+                </div>
+              </div>
+            ) : (
+              <div className="side-hidder-info pad-t-1 pad-l-1">
+                <div
+                  onClick={(e) => {
+                    e.stopPropagation();
```

Comment:
Why you are using `e.stopPropagation()` ?

Reply:
That was not needed..I was using stopPropagation for some of the click handler...So Accidentally i was here too..